### ISSUE TITLE
Other servers should be faster notified about account suspensions

### ISSUE THREAD

**Issue by joenepraat** [mastodon/mastodon#33053]
### Steps to reproduce the problem

The last weeks there are many spam accounts on mastodon.social and other servers. When we receive reports about those spam accounts, we noticed that they are already suspended on (e.g.) mastodon.social, but not on our server yet. This is a problem, because our users are still victim of spam (until our server get notification from mastodon.social about those suspended accounts). If I remembered correctly remote servers are not notified about suspended accounts with a high priority (pull, push queues?). Therefore I think this should have a higher priority, so remote servers can immediately remove the spam. 

### Expected behaviour

Remote spam should be cleaned up immediately

### Actual behaviour

There is sometimes a long delay before remote servers are notified about suspended accounts, with the result that spam is not cleaned up that fast. Also resulting in extra moderation work on remote servers.

### Mastodon instance

todon.nl, todon.eu, mastodon.social

### Mastodon version

v4.3.1

### Technical details

All versions are up to date.

**Comment by ClearlyClaire** [comment]
What happens when a local account gets suspended is that:
- every server that has reported it is notified
- the servers that the suspended user has the most recently mentioned are notified
- upon deletion of the suspended account, usually 30 days later, every server known is notified

In particular, what is missing is that servers aware of the messages not because they were mentioned, but because they are following someone who was mentioned in a reply, are not notified. However, this is not an easy fix, as the suspended user's server does not know that these “bystanders” have seen the spam.

The fix for that probably lies in the mentioned servers forwarding the suspension notice along the same paths, but we have to be careful not to make that an easy avenue for reflection or amplification attacks, and be careful of possible infinite message loops as well.

**Comment by todoneunl** [comment]
@ClearlyClaire Thank you for the explanation. What I see is that users are reporting spam on one of our servers and when I check if the reported spam account is already suspended on the remote server, it mostly is. So somehow there is a delay in "every server that has reported it is notified".

See also https://circumstances.run/@agturcz/113543176733117448.

**Comment by ClearlyClaire** [comment]
Hm, it might be that it was reported *after* being dealt with at the source server. Notifying reporters happens at the moment the account is suspended by an admin.

**Comment by todoneunl** [comment]
> Hm, it might be that it was reported _after_ being dealt with at the source server. Notifying reporters happens at the moment the account is suspended by an admin.

Yes, that is of course possible and also proofs the problem when suspended spam accounts are not immediately propagated to remote servers.

**Comment by PrivateGER** [comment]
This continues to be a big problem. 

Mastodon is the only implementation I'm aware of that does inbox forwarding, which massively amplifies the reach of spam, but does not federate out deletion of said spam to the same context. If you cannot do reasonable spam prevention, inbox forwarding is an irresponsible feature to still have active.

**Comment by ClearlyClaire** [comment]
Recently-followed accounts should be made aware of the suspension since #34294 (which has been merged in 4.3.7 and 4.2.20).

As for inbox forwarding, the issue is that inbox forwarding for messages is pretty straightforward (if potentially expensive), but propagating suspensions is a different matter, as the set of inboxes you'd need to forward to is more expensive to compute (you have to look at arbitrary many posts that you have forwarded for) and may be orders of magnitude larger.
